cmd/virt-exportserver/virt-exportserver.go

```diff
@@ -57,6 +62,30 @@ func (vi volumeInfo) getHandlers() map[string]http.Handler {
 	if vi.dirURI != "" {
 		result[vi.dirURI] = dirHandler(vi.dirURI, vi.path)
 	}
+	if vi.rawURI != "" {
+		fi, err := os.Stat(vi.path)
+		if err != nil {
+			log.Log.Reason(err).Errorf("error statting %s", vi.path)
+		} else {
+			if fi.IsDir() {
+				result[vi.rawURI] = fileHandler(path.Join(vi.path, "disk.img"))
+			} else {
+				result[vi.rawURI] = fileHandler(vi.path)
+			}
+		}
+	}
+	if vi.rawGzURI != "" {
+		fi, err := os.Stat(vi.path)
+		if err != nil {
+			log.Log.Reason(err).Errorf("error statting %s", vi.path)
+		} else {
+			if fi.IsDir() {
+				result[vi.rawGzURI] = gzipHandler(path.Join(vi.path, "disk.img"))
+			} else {
+				result[vi.rawGzURI] = gzipHandler(vi.path)
+			}
+		}
```

Comment:
can we avoid the repetition of the if statement and evaluated it only once. Something like:
```go
        fi, err := os.Stat(vi.path)
        if err != nil {
                log.Log.Reason(err).Errorf("error statting %s", vi.path)
                return result
        } 

        path := vi.path
        if fi.IsDir() {
                path = path.Join(vi.path, "disk.img")
        }

        if vi.rawURI != ""  {
                result[vi.rawURI] = fileHandler(path)
        }
        if vi.rawGzURI != ""  {
                result[vi.rawGzURI] = gzipHandler(path)
        }
```